ast/Trees.h

```diff
@@ -178,9 +176,9 @@ class TreePtr {
         return static_cast<Tag>(value);
     }
 
-    Expression *get() const noexcept {
+    void *get() const noexcept {
```

Comment:
It'd be nice if this were private or otherwise restricted from use.

Reply:
Yes, it would be. There are parts of the codebase that still rely on being able to use pointer equality for things like map keys, so it'll be some work to make this private.